plugins/toolbar/prism-toolbar.js

```diff
@@ -0,0 +1,82 @@
+(function(){
+	if (typeof self === 'undefined' || !self.Prism || !self.document) {
+		return;
+	}
+
+	var callbacks = [];
+
+	Prism.plugins.toolbar = {};
+
+	/**
+	 * Register a button callback with the toolbar.
+	 *
+	 * @param {Object|Function} opts
+	 */
+	Prism.plugins.toolbar.registerButton = function (opts) {
+		var callback;
+
+		if (typeof opts === 'function') {
+			callback = opts;
+		} else {
+			callback = function (env) {
+				var element;
+
+				if (typeof opts.onClick === 'function') {
+					element = document.createElement('a');
+					element.addEventListener('click', function () {
+						opts.onClick(env);
```

Comment:
I would also pass the `this` pointer like this:

```js
opts.onClick.call(this, env, opts);
```

That way the callback function gets access to the element that was clicked.